docs/source/udf.rst

```diff
@@ -64,16 +66,31 @@ to a single value:
    :code:
 
 
-Here is another example, demonstrating `kind="sig"` buffers and the merge function:
+Here is another example, demonstrating :code:`kind="sig"` buffers and the merge function:
 
 
 .. include:: udf/max.py
    :code:
 
 
-For a more complete example, please have a look at the functions implemented in `libertem.udf`,
-for example `blobfinder`.
+For a more complete example, please have a look at the functions implemented in the sub-modules of :code:`libertem.udf`,
+for example :code:`libertem.udf.blobfinder`.
+
+Auto UDF
+--------
+
+The :class:`AutoUDF` class and :meth:`Context.map()` method allow to run simple functions that accept a frame as the only parameter with an auto-generated :code:`kind="nav"` result buffer over a dataset ad-hoc without defining an UDF class. For more advanced processing, such as custom merge functions, postprocessing or performance optimization through tiled processing, defining an UDF class is required.
+
+The :class:`AutoUDF` class determines the output shape and type by calling the function with a mock-up frame of the same type and shape as a real detector frame and converting the return value to a numpy array. The :code:`extra_shape` and :code:`dtype` parameters for the result buffer are derived automatically from this numpy array. Additional constant parameters can be passed to the function via :meth:`functools.partial`, for example. The return value should be much smaller than the input size for this to work efficiently.
 
+Example: Calculate sum over the last signal axis.
+
+.. code-block:: python
+
+      result = ctx.map(
```

Comment:
Maybe the equivalent `AutoUDF(f=f)` plus `run_udf` call should also be documented here? Otherwise it makes little sense to even talk about `AutoUDF`, which then becomes only an implementation detail.